packages/api/src/routes/ehr/athenahealth/patient.ts

```diff
@@ -0,0 +1,34 @@
+import Router from "express-promise-router";
+import httpStatus from "http-status";
+import { Request, Response } from "express";
+import { getPatient } from "../../../external/ehr/athenahealth/command/get-patient";
+import { requestLogger } from "../../helpers/request-logger";
+import { asyncHandler, getCxIdOrFail, getFrom } from "../../util";
+import { getAuthorizationToken } from "../../util";
+
+const router = Router();
+
+/**
+ * GET /ehr/athena/patient/:id
+ *
+ * Retrieves the organization with the specified OID from CommonWell.
+ * @param req.params.oid The OID of the organization to retrieve.
+ * @returns Returns the organization with the specified OID.
+ */
+router.get(
+  "/:id",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const accessToken = getAuthorizationToken(req);
+    const cxId = getCxIdOrFail(req);
+    const athenaPatientId = getFrom("params").orFail("id", req);
```

Comment:
Are we doing authorization somewhere (checking this token can access this cxId and patientId? Can you link to it if so, please?

Reply:
Whether or not the token can access the cxId is done in the middleware here https://github.com/metriport/metriport/pull/2665/files#diff-cd228203ad07adb35f97c456351769ae7732e2495bcd6783c2c062b98b00b088

When we onboard an Athena customer, we will add their practice ids to the cx mapping table so that they map to the corresponding Metriport customer (i.e. `a-1.Practice-1234567` maps to cxId `XXX`). When we generate the Authorization token, we get back the associated practice id **from Athena** that we will save with the token. When this customer makes a request, we map the token to the saved practice id, which then maps to the cxId.

We only authorize access to the Athena patient on the actual request -- if the incoming Authorization token isn't valid to access the Athena patient, we simply won't find an Athena patient, and subsequently find no Metriport patient.